### PR TITLE
Ternary Search Algorithm Added

### PR DESCRIPTION
**Pull request by oldMarcosVillacanas**
Ternary Search was missing, now is available. 
The code has a main function to be used as a test, for example, in https://play.golang.org/. 
I have added some comments to make it easier to understand.

### CODE REVIEW

File: searches/ternary_search.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	// First boundary: between lowIndex and first 1/3 of array
+	mid1 := int(lowIndex + (highIndex-lowIndex)/3)
+
+	// Second boundary: between first 1/3 and second 1/3 of array
+	mid2 := int(lowIndex + 2*(highIndex-lowIndex)/3)
+
+	if target == array[mid1] {
+		// Value found in first boundary position
+		return mid1
+	} else if target == array[mid1] {
```
[the remark this answers is not in the document]
Comment (MarcosVillacanas) [reply searches/ternary_search.go]: Solved, thank you

File: searches/ternary_search.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
+/*
+	Ternary search implementation in Go
+
+	    Worst-case performance	Θ(log3(N))
```
[the remark this answers is not in the document]
Comment (MarcosVillacanas) [reply searches/ternary_search.go]: Solved on new commit, thank you

**Comment by luislizarraga** [comment]
Looks good, great work!

File: searches/ternary_search.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ 		Average performance	    Θ(log(N))
+*/
+package main
+
+import (
+	"fmt"
+	"strconv"
+)
+
+func main() {
```
Comment (nomisrevol) [inline searches/ternary_search.go:15]: Remove `main` function and add a new test file instead.

**Review by nomisrevol — CHANGES_REQUESTED** [review]

**Comment by tjgurwara99** [comment]
Please take a look at our new [contribution guidelines](https://github.com/TheAlgorithms/Go/blob/d2f12849df76203d80762ceca5105531d3424384/CONTRIBUTING.md) and make changes accordingly. Currently this PR is not up to standard. If no changes are made in the next two weeks, I will consider this PR to be stale and will close this PR then.

**Pull request closed without merging**